pkg/talos/networkconfig.go

```diff
@@ -2,15 +2,72 @@ package talos
 
 import (
 	"bytes"
+	"fmt"
 	"net/netip"
+	"strings"
 
 	"github.com/budimanjojo/talhelper/v3/pkg/config"
 	"github.com/siderolabs/go-pointer"
+	"github.com/siderolabs/talos/pkg/machinery/cel"
+	"github.com/siderolabs/talos/pkg/machinery/cel/celenv"
 	"github.com/siderolabs/talos/pkg/machinery/config/types/network"
+	"github.com/siderolabs/talos/pkg/machinery/config/types/v1alpha1"
 	"github.com/siderolabs/talos/pkg/machinery/nethelpers"
 	"gopkg.in/yaml.v3"
 )
 
+func validateInterfaceNames(devices []*v1alpha1.Device) error {
+	interfaceTypes := map[string]string{}
+
+	for i, device := range devices {
+		if device.DeviceInterface == "" {
+			continue
+		}
+
+		var configTypes []string
+		if device.DeviceBond != nil {
+			configTypes = append(configTypes, "bond")
+		}
+		if device.DeviceBridge != nil {
+			configTypes = append(configTypes, "bridge")
+		}
+		if device.DeviceWireguardConfig != nil {
+			configTypes = append(configTypes, "wireguard")
+		}
+
+		// A device can only have ONE primary config type (bond, bridge, or wireguard)
```

Comment:
Is this true? The upstream `v1alpha1.ValidateNetworkDevices` doesn't check for this so I'm not sure.